src/platform/PICO/bus_quadspi_pico.c

```diff
@@ -188,26 +192,80 @@ static void quadSpiDmaSegment(const extDevice_t *dev, busSegment_t *segment)
         return;
     }
 
+    // Buffer to hold the full 32 bit words so x2 and x4 widths can be used
+    static uint32_t direct_tx32[QSPI_MAX_TX_BYTES];
+    uint32_t direct_tx32_reset = 0;
+    // If the wide flag is true use the direct_tx32 buffer, otherwise optimise
+    // and use the txData directly
+    bool wide = true;
+
+    // Set the direction, input if there is data to be received
+    if (!segment->u.buffers.rxData) {
+        direct_tx32_reset |= QMI_DIRECT_TX_OE_BITS;
+    }
+
+    switch (segment->len & BUS_SEGMENT_LEN_WIDTH_MASK) {
+    case BUS_SEGMENT_LEN_WIDTH_X1:
+    default:
+        direct_tx32_reset |= QMI_DIRECT_TX_IWIDTH_VALUE_S << QMI_DIRECT_TX_IWIDTH_LSB;
+        wide = false;
+        break;
+
+    case BUS_SEGMENT_LEN_WIDTH_X2:
+        direct_tx32_reset |= QMI_DIRECT_TX_IWIDTH_VALUE_D << QMI_DIRECT_TX_IWIDTH_LSB;
+        break;
+
+    case BUS_SEGMENT_LEN_WIDTH_X4:
+        direct_tx32_reset |= QMI_DIRECT_TX_IWIDTH_VALUE_Q << QMI_DIRECT_TX_IWIDTH_LSB;
+        break;
+
+    }
+
+    if (wide) {
+        // Initialise the 32 bit buffer
+        if (segment->u.buffers.txData) {
+            if ((segment->len & BUS_SEGMENT_LEN_LENGTH_MASK) <= QSPI_MAX_TX_BYTES) {
+                for (int i = 0; i < (segment->len & BUS_SEGMENT_LEN_LENGTH_MASK); i++) {
+                    direct_tx32[i] = direct_tx32_reset | segment->u.buffers.txData[i];
+                }
+            } else {
+                // This isn't at all graceful, but it will be pretty evident when a write transfer fails completely.
+                // It's known that blackbox never uses a buffer more than 65 bytes, and writing over USB isn't supported.
+                // USB reads are unaffected.
+                // Abort the transaction
+                bus->curSegment = (busSegment_t *)BUS_QSPI_FREE;
+                // Negate CS
+                qmi_cs1_assert(dev, false);
+                return;
+            }
+        } else {
+            dummyTx = direct_tx32_reset;
+        }
+    } else {
+        // Default value for tx bytes
+        dummyTx = 0;
+    }
+
     // Launch next segment via DMA
     dmaChannelDescriptor_t *dmaRx = bus->dmaRx;
     dmaChannelDescriptor_t *dmaTx = bus->dmaTx;
 
     // Use the correct callback argument
     dmaRx->userParam = (uint32_t)dev;
 
-    quadSpiInternalInitStream(bus, segment->u.buffers.txData, segment->u.buffers.rxData);
+    quadSpiInternalInitStream(bus, segment->u.buffers.txData, segment->u.buffers.rxData, wide);
```

Comment:
txData / rxData is used only as boolean and different txData value is passed (raw data, not expanded). It will be cleaner to pass bool (or real tx buffer)